### DISCUSSION TITLE
[archived] "Workflows Variables" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#12523] [General]
# en/latest/variables/

https://argo-workflows.readthedocs.io/en/latest/variables/

**Comment by sampathbalsu2** [comment]
`workflow.failures`
>	A list of JSON objects containing information about nodes that failed or errored during execution. Available fields: displayName, message, templateName, phase, podName, and finishedAt

in the above, message value is some times blank, and some times its is coming as "exit code 127". do we have the option to display the actual failure message from the step?

**Comment by artemklevtsov** [comment]
Where can I get more information about jsonpath function? Syntax etc.

**Reply by agilgur5** [reply]
JSONPath has [a specification](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html) that is not unique to Argo and is used by many tools.

Argo specifically [uses](https://github.com/argoproj/argo-workflows/blob/0ca0c0f721c9bf05c63dea98168cfa87441656c7/go.mod#L89) the implementation from the library [`evilmonkeyinc/jsonpath`](https://github.com/evilmonkeyinc/jsonpath) at time of current writing.
